src/Config/ObjectModel/RuntimeConfig.cs

```diff
@@ -19,6 +20,14 @@ public record RuntimeConfig
 
     public RuntimeEntities Entities { get; init; }
 
+    public DataSourceFiles? DataSourceFiles { get; init; }
+
+    [JsonIgnore]
+    public bool CosmosEngineNeeded { get; private set; }
+
+    [JsonIgnore]
+    public bool SqlEngineNeeded { get; private set; }
```

Comment:
these names aren't really informative. `SqlEngineNeeded` for what? 

In other code, it seems that this indicates whether mutation/query engines need to be built for the respective dbtypes. Instead of `<X>EngineNeeded` would `<X>DataSourceSet` or `<X>DataSourceUsed` be more generic?

That way, dependent logic can just call:
```csharp
if (XDataSourceSet)
{
    /// Create engines/executors/miscellaneous services.
}
```

Reply:
I interpret `SqlEngineNeeded` to be a flag that will be reset upon usage of checking the value. e.g. create the 'engine' and then flip that flag to false.

Reply:
I think DataSourceUsed makes more sense. i will make this change.